pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -273,33 +278,47 @@ protected void handleLookup(CommandLookupTopic lookup) {
     @Override
     protected void handlePartitionMetadataRequest(CommandPartitionedTopicMetadata partitionMetadata) {
         final long requestId = partitionMetadata.getRequestId();
-        final String topicName = partitionMetadata.getTopic();
         if (log.isDebugEnabled()) {
-            log.debug("[{}] Received PartitionMetadataLookup from {} for {}", topicName, remoteAddress, requestId);
+            log.debug("[{}] Received PartitionMetadataLookup from {} for {}", partitionMetadata.getTopic(),
+                    remoteAddress, requestId);
         }
+
+        DestinationName topicName;
+        try {
+            topicName = DestinationName.get(partitionMetadata.getTopic());
+        } catch (Throwable t) {
+            if (log.isDebugEnabled()) {
+                log.debug("[{}] Failed to parse topic name '{}'", remoteAddress, partitionMetadata.getTopic(), t);
+            }
+            ctx.writeAndFlush(newLookupErrorResponse(ServerError.InvalidTopicName,
+                    "Invalid topic name: " + t.getMessage(), requestId));
```

Comment:
ctx.writeAndFlush(
                                Commands.newPartitionMetadataResponse(ServerError.InvalidTopicName, msg, requestId));

Should not be newLookupErrorResponse